Mathlib/Order/SuccPred/Archimedean.lean

```diff
@@ -396,3 +396,117 @@ instance Set.OrdConnected.isSuccArchimedean [SuccOrder α] [IsSuccArchimedean α
   inferInstanceAs (IsSuccArchimedean sᵒᵈᵒᵈ)
 
 end OrdConnected
+
+section Monotone
+variable {α β : Type*} [PartialOrder α] [Preorder β]
+
+section SuccOrder
+variable [SuccOrder α] [IsSuccArchimedean α] {s : Set α} {f : α → β}
+
+lemma monotoneOn_of_le_succ (hs : s.OrdConnected)
+    (hf : ∀ a, ¬ IsMax a → a ∈ s → succ a ∈ s → f a ≤ f (succ a)) : MonotoneOn f s := by
```

Comment:
Can you golf this using `monotone_iff_forall_covBy`?

Reply:
Hmm, I would first need to show that an archimedean succ order is locally finite and prove the `On` versions of the lemmas there. And it would result in an import increase. Not sure it's worth it?